### ISSUE TITLE
Add support for Model X door controls

### ISSUE THREAD

**Issue by lytesout2376** [timdorr/tesla-api#600]
Refreshed Model X - front and falcon wing doors can now be controlled via the Tesla app. This is shown in Tesla's official tutorial video that can be found here: https://youtu.be/jhbhxC7oFzM

Can this be added to the documentation?

**Comment by core-hacked** [comment]
Sad that it's for the refresh only, however I haven't seen any new endpoints for this in the endpoints file, this could mean that it's either not included, in another endpoints file (or something of that sort), or it is part of the `/api/1/vehicles/{id}/command/door_unlock` endpoint, maybe another field in the post body that indicates the door?

Something like: 
```json
{
  "door": 0-3,
  "state": 0-1 or true/false maybe even "open"/"closed"
}
```
or maybe 
```json
{
  "door": 0-3,
  "open": true/false or "closed": true/false
}
```
It might also just be a new endpoint like `/api/1/vehicles/{id}/command/falcon_door_unlock`
and a `"side":0-1` or `"side": "left/right"` in the post body

Maybe even `/api/1/vehicles/{id}/command/mx_door_unlock`?

**Comment by itsMeDavidV** [comment]
Bump, just giving this issue a bit more life. Anyone discover an endpoint for this as of late?

**Comment by core-hacked** [comment]
Bump, I am currently looking into this, but I am limited since I do not own a refreshed Model X. 
I hope to find something, maybe someone with a refreshed model x can help?

**Comment by core-hacked** [comment]
Hey @lytesout2376 can you try to call the `door_unlock` endpoint with a post body parameter called door and the value of it being one of the following 2 character strings:
```css
driver_front = df
driver_rear = dr
passenger_front = pf
passenger_rear = pr
```
These were pulled by decompiling the Tesla Apk. It seems to be the door_unlock endpoint.

**Comment by lytesout2376** [comment]
@core-hacked thanks, I just tried it. On first call here's the response I got:

> {"response":{"result":false,"reason":"could_not_wake_buses"}}

I have a legacy X. It seems the refreshed X that has this capability in the Tesla app have buses that control these doors?

On subsequent calls after, this is the response I get:

> {"response":{"result":true,"reason":""}}

Which, according to the API docs is the expected response when unlocking the vehicle. Also worth noting that when I get this response, it unlocks the vehicle even though a body parameter "door" with any of the 2 character strings passed along with it.

So in conclusion, we need to have someone test this on a refreshed S/X. It seems the legacy S/X are left out on this as it requires additional hardware buses.

**Comment by core-hacked** [comment]
@lytesout2376 that's the sad part about tesla, the feature could just be done with software since inside the car the UI can call a specific door, all tesla needs to do is add it for legacy cars. Basically: App request > car > UI_OpenDriverDoor > door opens

**Comment by core-hacked** [comment]
As mentioned in #637, after extensive digging, the endpoint does not exist. It's controlled via bluetooth.
